### PR TITLE
feat(discovery): support endpointslices in kubernetes discovery

### PR DESCRIPTION
**Pull request by dongjiang1989**
### Description

This PR add support endpointslices in kubernetes discovery

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/discovery/kubernetes/init.lua [hunk 1]
```diff
@@ -367,8 +430,12 @@ local function single_mode_init(conf)
     end
 
     local default_weight = conf.default_weight
-
-    local endpoints_informer, err = informer_factory.new("", "v1", "Endpoints", "endpoints", "")
+    local endpoints_informer, err
+	if conf.watch_endpoint_slices_schema then
```
Comment (dongjiang1989) [inline apisix/discovery/kubernetes/init.lua:434]:
```suggestion
    if conf.watch_endpoint_slices_schema then
```

**Comment by dongjiang1989** [comment]
> @dongjiang1989 please explain why this is needed, thanks.

@shreemaan-abhishek thanks for your review.

EndpointSlices provide a simple way to track network endpoints within a Kubernetes cluster. They offer a more scalable and extensible alternative to Endpoints.

ref: https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/

**Comment by dongjiang1989** [comment]
@monkeyDluffy6017  @shreemaan-abhishek  Please re-check. Thanks

File: t/kubernetes/discovery/kubernetes.t [hunk 2]
```diff
@@ -128,6 +128,7 @@ GET /compare
   "client": {
     "token": "${KUBERNETES_CLIENT_TOKEN}"
   },
+  "watch_endpoint_slices": false,
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply t/kubernetes/discovery/kubernetes.t:131]: Thanks for your review. This just check default value in response body

File: apisix/discovery/kubernetes/informer_factory.lua [hunk 3]
```diff
@@ -355,7 +355,7 @@ function _M.new(group, version, kind, plural, namespace)
     end
 
     if namespace and namespace ~= "" then
-        path = path .. "/namespace/" .. namespace
+        path = path .. "/namespaces/" .. namespace
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply apisix/discovery/kubernetes/informer_factory.lua:358]: Thanks for your review. It's must for e2e case. Squash and merge in last pull merge

[the remark this answers is not in the document]
Comment (shreemaan-abhishek) [reply apisix/discovery/kubernetes/informer_factory.lua:358]: > It's must for e2e case. Squash and merge in last pull merge

I don't understand you but let me clarify myself. A PR should only contain related changes. If this PR depends upon another feature/bug fix. The other feature should be merged first.

[the remark this answers is not in the document]
Comment (dongjiang1989) [reply apisix/discovery/kubernetes/informer_factory.lua:358]: Yes. Merge #10917 pr frist. Then update this PR.

**Comment by monkeyDluffy6017** [comment]
@zhixiongdu027 Could you help to review this pr?

**Comment by AlinsRan** [comment]
We should update the documentation for this feature.

**Comment by dongjiang1989** [comment]
> We should update the documentation for this feature.

Thanks. Done.

**Comment by dongjiang1989** [comment]
@AlinsRan @shreemaan-abhishek @monkeyDluffy6017 please re-check，thanks

**Review by AlinsRan — APPROVED** [review]

**Comment by liugang594** [comment]
@dongjiang1989 
咨询一下，咱们这个endpointslice的支持，最终得到服务节点列表是全量的么？看代码中像是只得到了最后变更的slice的pod ip列表。

**Pull request merged**